Sources/StreamChat/APIClient/CDNClient/CDNClient.swift

```diff
@@ -18,6 +24,34 @@ public protocol CDNClient {
         progress: ((Double) -> Void)?,
         completion: @escaping (Result<URL, Error>) -> Void
     )
+
+    /// Uploads attachment as a multipart/form-data and returns the uploaded remote file and its thumbnail.
+    /// - Parameters:
+    ///   - attachment: An attachment to upload.
+    ///   - progress: A closure that broadcasts upload progress.
+    ///   - completion: Returns the uploaded file's information.
+    func uploadAttachment(
+        _ attachment: AnyChatMessageAttachment,
+        progress: ((Double) -> Void)?,
+        completion: @escaping (Result<UploadedFile, Error>) -> Void
+    )
```

Comment:
This one is similar to the one in the `AttachmentUploader` protocol 🤔 It feels a bit odd, but not sure if we have any other alternative.

Reply:
Yeah, could not find a better way... 🙃 The part that worries me the most is that it can give "ambiguous usage" errors, as the only difference is the Result.Success type.

Maybe the method signature should be a bit different?

Reply:
Hum, I think it is fine to have the same name for now 🤔

Reply:
What about deprecating the other one? @polqf